apps/full-site-editing/full-site-editing-plugin/editor-site-launch/src/launch-steps/name-step/index.tsx

```diff
@@ -16,29 +17,56 @@ import LaunchStepContainer, { Props as LaunchStepProps } from '../../launch-step
 import './styles.scss';
 
 const NameStep: React.FunctionComponent< LaunchStepProps > = ( { onNextStep } ) => {
-	const { setStepComplete } = useDispatch( LAUNCH_STORE );
+	const domain = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedDomain() );
+	const { setStepComplete, setStepIncomplete, setDomainSearch } = useDispatch( LAUNCH_STORE );
+	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
 
-	const handleClick = () => {
-		setStepComplete( LaunchStep.Name );
+	const handleSave = () => {
+		saveEditedEntityRecord( 'root', 'site' );
+
+		// update domainSearch only if there is no custom Domain selected
+		! domain && setDomainSearch( title );
+
+		title.length ? setStepComplete( LaunchStep.Name ) : setStepIncomplete( LaunchStep.Name );
+	};
+
+	const handleNext = () => {
+		handleSave();
 		onNextStep?.();
 	};
 
 	return (
 		<LaunchStepContainer className="nux-launch-name-step">
 			<div className="nux-launch-step__header">
 				<div>
-					<Title>{ X__( 'Name your site', 'full-site-editing' ) }</Title>
-					<SubTitle>{ X__( 'Pick a name for your site.', 'full-site-editing' ) }</SubTitle>
+					<Title>{ __( 'Name your site', 'full-site-editing' ) }</Title>
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.